### PR TITLE
deprecation: move `types.d.ts` to `std/io`

### PR DESCRIPTION
**Pull request by iuioiua**
This change is required for workspaces support.

### CODE REVIEW

File: _tools/check_circular_submodule_dependencies.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ async function check(
     }
   }
   deps.delete(submod);
-  deps.delete("types.d.ts");
+  deps.delete("io/types.d.ts");
```
Comment (lucacasonato) [inline _tools/check_circular_submodule_dependencies.ts:41]: I think you should remove this line instead.

**Review by lucacasonato — COMMENTED** [review]
Looks good

File: _tools/check_circular_submodule_dependencies.ts (hunk 1 above)
Comment (kt3k) [reply _tools/check_circular_submodule_dependencies.ts:41]: That's correct

Comment (iuioiua) [reply _tools/check_circular_submodule_dependencies.ts]: Fixed

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**